### PR TITLE
Hastexo/master/openstack

### PR DESCRIPTION
**Pull request by arbrandes**

### CODE REVIEW

File: heat-templates/README.md [hunk 1]
```diff
@@ -198,47 +198,91 @@ parameters when invoking Heat:
...
+certs_version: "master"
+forum_source_repo: "https://{{ COMMON_GIT_MIRROR }}/hastexo/edx-forum.git"
+forum_version: "master"
+xqueue_source_repo: "https://{{ COMMON_GIT_MIRROR }}/hastexo/edx-xqueue.git"
+xqueue_version: "hastexo/integration/hastexo"
+NOTIFIER_SOURCE_REPO: "https://{{ COMMON_GIT_MIRROR }}/hastexo/edx-notifier.git"
+NOTIFIER_VERSION: "master"
+```
+
+At this point, make sure to create a Swift bucket for this deployment: it'll be
```
Comment (fghaas) [inline heat-templates/README.md]: Swift doesn't have buckets, please replace this with "container".

File: heat-templates/README.md [hunk 2]
```diff
@@ -198,47 +198,91 @@ parameters when invoking Heat:
...
+needed during the initial playbook run.  You can do so from your own computer,
+as long as the `swift` command line client is installed and the appropriate
+OpenStack credentials are loaded into the environment.  By default, the sample
+Ansible variables will configure edX to use the "lms.example.com" bucket.  Run
+the following to create the bucket with read permissions for connections
+originating fromt he site itself:
+
+```
+SITE=lms.example.com; swift post -r ".r:$SITE" $SITE
```
Comment (fghaas) [inline heat-templates/README.md]: In this case, the `openstack` client command syntax is actually more intuitive and self-explanatory than the one for `swift`. Suggest to replace `swift post` with `openstack container create`.

File: heat-templates/README.md (hunk 1 above)
Comment (fghaas) [inline heat-templates/README.md]: Also, this paragraph here could be automated with an [`os_object`](http://docs.ansible.com/ansible/os_object_module.html) task in a playbook. However, this would require that deployers use Ansible 2.x, whereas edX still seems stuck on their weird 1.9.3 fork. Is that the reason why the manual steps are described here?

File: heat-templates/README.md [hunk 3]
```diff
@@ -340,25 +395,22 @@ node cluster, go back to your deploy node and:
...
-   appropriately:
-
-    ```
-    $ cd /var/tmp/edx-configuration/playbooks
+    $ cd ~/edx-configuration/playbooks
     $ vim openstack-multi-node.yaml
     ...
     - certs
     - demo
+    - oauth_client_setup
```
Comment (fghaas) [inline heat-templates/README.md]: Why is `oauth_client_setup` mentioned here?

File: playbooks/openstack/host_vars/localhost.example [hunk 4]
```diff
@@ -241,12 +245,14 @@ EDXAPP_JWT_AUDIENCE: "lms-key"
 EDXAPP_JWT_SECRET_KEY: "lms-secret"
 
 # Insights OAuth2
-INSIGHTS_LMS_BASE: '{{ EDXAPP_LMS_ROOT_URL }}'
```
Comment (fghaas) [inline playbooks/openstack/host_vars/localhost.example]: These Insights-related changes look out ouf scope for this PR. Did the original values break anything?

File: playbooks/openstack/host_vars/localhost.example [hunk 5]
```diff
@@ -288,3 +294,66 @@ EDXAPP_XBLOCK_SETTINGS:
 gateone_user_dir: "{{COMMON_DATA_DIR }}/edxapp/terminal_users"
 gateone_url_prefix: "/terminal"
 gateone_port: 28010
+
+# Log sync
```
Comment (fghaas) [inline playbooks/openstack/host_vars/localhost.example]: Why are we adding all these log sync settings here?

File: playbooks/openstack/group_vars/all.example [hunk 6]
```diff
@@ -14,11 +14,37 @@ xqueue_version: "master"
 NOTIFIER_SOURCE_REPO: "https://{{ COMMON_GIT_MIRROR }}/edx/notifier.git"
 NOTIFIER_VERSION: "master"
 
-# Cluster
-MONGO_CLUSTER_KEY: 'password'
+# Keepalived
 keepalived_shared_ip: "192.168.122.110"
 
+# Mongo
```
Comment (fghaas) [inline playbooks/openstack/group_vars/all.example]: What about the Mongo values here? Did the prior omission of these break the playbooks?

File: heat-templates/README.md (hunk 1 above)
Comment (arbrandes) [inline heat-templates/README.md]: +1 for containers vs containers, but the primary reason the steps are manual is that I'm not yet convinced automating this is a good idea, necessarily.  It would require OpenStack credentials be made available to the Ansible playbooks, to start with.   And yes, edX's still on 1.9.3.

File: heat-templates/README.md (hunk 2 above)
Comment (arbrandes) [inline heat-templates/README.md]: Good idea.  Didn't even know `openstack` could do Swift.

File: heat-templates/README.md (hunk 3 above)
Comment (arbrandes) [inline heat-templates/README.md]: It's just a manual diff.  The current playbook includes it, so I included it here.

File: playbooks/openstack/host_vars/localhost.example (hunk 4 above)
Comment (arbrandes) [inline playbooks/openstack/host_vars/localhost.example]: Yes, the sample Insights stuff was breaking stuff.  Not this particular line, but one below.

File: playbooks/openstack/host_vars/localhost.example (hunk 5 above)
Comment (arbrandes) [inline playbooks/openstack/host_vars/localhost.example]: I'm trying to keep the sample group/host vars as close as possible to our private version, to simplify maintenance.

File: playbooks/openstack/group_vars/all.example (hunk 6 above)
Comment (arbrandes) [inline playbooks/openstack/group_vars/all.example]: See above.  Just syncing up with ours.

**Comment by arbrandes** [comment]
PR amended!

File: playbooks/openstack/group_vars/all.example (hunk 6 above)
Comment (fghaas) [inline playbooks/openstack/group_vars/all.example]: Yes, I get that, but can we have those in separate commits if it's not too much trouble?

File: playbooks/openstack/host_vars/localhost.example (hunk 5 above)
Comment (fghaas) [inline playbooks/openstack/host_vars/localhost.example]: Could that also be in a separate commit please?
